src/main/java/com/edmunds/rest/databricks/JobRunner.java

```diff
@@ -81,12 +81,18 @@ public void runJob() throws IOException, DatabricksRestException, InterruptedExc
   private JobService getService() {
     String username = parser.getUsername();
     String password = parser.getPassword();
+    String token = parser.getToken();
     String hostname = parser.getHostname();
 
-    DatabricksServiceFactory factory =
-        DatabricksServiceFactory.Builder
-            .createUserPasswordAuthentication(username, password, hostname)
-            .build();
+    DatabricksServiceFactory factory = null;
+    if (token != null) {
+      factory = DatabricksServiceFactory.Builder.createTokenAuthentication(token, hostname)
+          .build();
+    } else {
+      DatabricksServiceFactory.Builder
```

Comment:
It looks like you missed the factory assignment on line 92, which, I think would cause a NPE.

Reply:
oh yikes.

Without any testing for user-password,(I removed it all) this was not caught.